### PR TITLE
feat(Dataworker): ZkSync should count ETH and WETH towards balance needed to execute leaves

### PR DESCRIPTION
**Pull request by nicholaspai**

### CODE REVIEW

File: src/clients/BalanceAllocator.ts [hunk 1]
```diff
@@ -141,9 +142,19 @@ export class BalanceAllocator {
...
   }
 
+  isEthAddress(chainId: number, tokenAddress: string): boolean {
+    // If there is an ETH address defined in CONTRACT_ADDRESSES, use it, otherwise assume ETH address is the zero
+    // address.
+    if (isDefined(CONTRACT_ADDRESSES[chainId]?.eth?.address)) {
+      return CONTRACT_ADDRESSES[chainId].eth.address === tokenAddress;
+    } else {
+      return ZERO_ADDRESS === tokenAddress;
+    }
```
Comment (james-a-morris) [inline src/clients/BalanceAllocator.ts:152]:
```suggestion
    return ( CONTRACT_ADDRESSES[chainId]?.eth?.address ?? ZERO_ADDRESS ) === tokenAddress;
```

File: src/dataworker/DataworkerUtils.ts [hunk 2]
```diff
@@ -387,3 +389,26 @@ export async function _buildPoolRebalanceRoot(
     tree: buildPoolRebalanceLeafTree(leaves),
   };
 }
+
+// Some SpokePools will contain balances of ETH and WETH, while others will only contain balances of WETH,
+// so if the l2TokenAddress is WETH and the SpokePool is one such chain that holds both ETH and WETH,
+// then it should return both ETH and WETH. For other chains, it should only return the l2TokenAddress.
+function getWethAndEth(chainId): string[] {
+  const wethAndEth = [CONTRACT_ADDRESSES[chainId].weth.address, CONTRACT_ADDRESSES[chainId].eth.address];
+  if (wethAndEth.some((chainId) => !isDefined(chainId))) {
```
Comment (james-a-morris) [inline src/dataworker/DataworkerUtils.ts:398]: Is `chainId` supposed to be a chain id or a token address?

**Review by james-a-morris — COMMENTED** [review]
Left comments

File: src/dataworker/DataworkerUtils.ts (hunk 2 above)
Comment (nicholaspai) [reply src/dataworker/DataworkerUtils.ts:401]: ah yes it should be a token address, will rename

**Pull request merged**